sentimental/src/main/java/com/google/sps/data/DatastoreService.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.data;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatastoreService {
+
+  /** Returns a List of all of the Tweets stored in Datastore, newest first. */
+  public List<Tweet> getAllTweets() {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query =
+      Query.newEntityQueryBuilder().setKind("Tweet").setOrderBy(OrderBy.desc("timestamp")).build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    List<Tweet> tweets = new ArrayList<>();
+    while (results.hasNext()) {
+      Entity entity = results.next();
+
+      long id = entity.getKey().getId();
+      String text = entity.getString("text");
+      long timestamp = entity.getLong("timestamp");
+
+      Tweet tweet = new Tweet(id, text, timestamp);
+      tweets.add(tweet);
+    }
+
+    return tweets;
+  }
+
+  /**
+   * Saves the Tweets in Datastore. Tweets with duplicate IDs will not be
+   * repeated, so this is safe to call with the same Tweets multiple times.
+   */
+  public void saveTweets(List<Tweet> tweets) {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Tweet");
+
+    for (Tweet tweet : tweets) {
+      FullEntity taskEntity =
+          Entity.newBuilder(keyFactory.newKey(tweet.getID()))
+              .set("text", tweet.getText())
+              .set("timestamp", tweet.getTimestamp())
+              .build();
+      datastore.put(taskEntity);
+    }
```

Comment:
Maybe the indentation could be fixed to be uniform through all the file

Reply:
I know this looks a little ugly, but this is actually the indentation recommended by the Google style guide. A couple things are happening here:

First, the style guide recommends limiting line length to 100 characters: https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

It recommends wrapping long lines: https://google.github.io/styleguide/javaguide.html#s4.5-line-wrapping

It's not clear on exactly where you should put newlines, but it does say that when you wrap to the next line, you should then indent by 4 spaces instead of 2: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

All of that combines to create this kinda weird-looking indentation, but it is actually what the style guide suggests.